dali/test/python/test_pipeline.py

```diff
@@ -1395,6 +1397,62 @@ def test_skip_cached_images():
                           CachedPipeline(reader_type, batch_size, is_cached=True, skip_cached_images=True),
                           batch_size=batch_size, N_iterations=100)
 
+def test_caffe_no_label():
+    class CaffePipeline(Pipeline):
+        def __init__(self, batch_size, path_to_data, labels, seed=123456, skip_cached_images=False, num_shards=1):
+            super(CaffePipeline, self).__init__(batch_size, num_threads=1, device_id=0, prefetch_queue_depth=1, seed=seed)
+            self.input = ops.CaffeReader(path = path_to_data,
+                                          shard_id = 0,
+                                          num_shards = num_shards,
+                                          stick_to_shard = True,
+                                          prefetch_queue_depth = 1,
+                                          label_available = labels)
+            self.decode = ops.ImageDecoder(output_type = types.RGB)
+            self.labels = labels
+
+        def define_graph(self):
+            if not self.labels:
+                jpegs = self.input()
+            else:
+                jpegs,_ = self.input()
+            images = self.decode(jpegs)
+            return (images)
+
+    pipe = CaffePipeline(2, caffe_db_folder, True)
```

Comment:
you could use compare_pipelines to compare that the outputs are the same

Reply:
Not really, no_label data set is a small subset of the bigger one. I though there is no point in putting all images there just to check if it works without labels.